### ISSUE TITLE
Bug: Type error "{ icon: IconDefinition } is not assignable to type 'ComponentProps<ComponentPublicInstance<FontAwesomeIconProps, {}, {}, {}, {}, {}, FontAwesomeIconProps, {}, false, ComponentOptionsBase<any, any, any, any, any, any, any, any, any, {}>>>"

### ISSUE THREAD

**Issue by daniloribeiro00** [FortAwesome/Font-Awesome#19398]
### Bug description

I'm having an odd issue after updating to v6 in Vue.js.
It seems VSCode (with Volar extension) does not recognise the FontAwesomeIcon component types.

It shows the following message but it works normally on the browser without any warning on the console.
`"{ icon: IconDefinition } is not assignable to type 'ComponentProps<ComponentPublicInstance<FontAwesomeIconProps, {}, {}, {}, {}, {}, FontAwesomeIconProps, {}, false, ComponentOptionsBase<any, any, any, any, any, any, any, any, any, {}>>>'."`

Code below:
```vue
<script setup lang="ts">
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import { faFileCode, faHeadset } from '@fortawesome/free-solid-svg-icons';
</script>

<template>
  <div>
     <FontAwesomeIcon :icon="faFileCode" />
  </div>
</template>
```

package.json
```
    "@fortawesome/fontawesome-svg-core": "^6.2.0",
    "@fortawesome/free-solid-svg-icons": "^6.2.0",
    "@fortawesome/vue-fontawesome": "^3.0.1",
    "typescript": "^4.8.4",
    "vue": "^3.2.41",
    "vite": "^3.1.8",
```

### Screenshots

![image](https://user-images.githubusercontent.com/68981163/197608278-7b97b782-d35d-4fe7-9afa-e8e636f54f68.png)

![image](https://user-images.githubusercontent.com/68981163/197608190-bbf193fe-5b0f-4c2d-b924-6f4669b28159.png)
(the image is in Portuguese)

### Font Awesome version

v6.2.0

### Serving

Self-hosted

### Implementation

SVG+JS

### Browser and Operating System

- Chrome 106 on Windows 11 (despite the problem is not on the browser)

### Web bug report checklist

- [ ] I have included a test case because my odds go _way_ up that the team can fix this when I do
- [X] I have [searched for existing issues](https://github.com/FortAwesome/Font-Awesome/issues) and to the best of my knowledge this is not a duplicate

**Comment by henningway** [comment]
I have the same problem after updating vue-tsc from 0.40.9 to 1.0.9.

```
        "@fortawesome/fontawesome-common-types": "^6.2.0",
        "@fortawesome/fontawesome-svg-core": "^6.2.0",
        "@fortawesome/vue-fontawesome": "^3.0.1",
        "typescript": "^4.8.4",
        "vite": "^3.2.2",
        "vue": "^3.2.41",
        "vue-tsc": "^1.0.9"
```

**Comment by daniloribeiro00** [comment]
Fixed in @fortawesome/vue-fontawesome v3.0.2

**Comment by tagliala** [comment]
thanks for the heads-up
